File: erp-finance-system/backend/finance-core-service/src/main/java/com/erp/finance/entity/PendingAction.java
package com.erp.finance.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "pending_actions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PendingAction {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 255)
    private String action;

    @Column(nullable = false, length = 100)
    private String module;

    @Column(nullable = false)
    private String priority; // HIGH, MEDIUM, LOW

    @Column(nullable = false)
    private LocalDate dueDate;

    @Column(nullable = false)
    private String status; // PENDING, IN_PROGRESS, COMPLETED

    @Column(length = 100)
    private String assignedTo;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDate createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDate.now();
    }
}

File: erp-finance-system/mobile/erp_finance_mobile/lib/models/pending_action.dart
class PendingAction {
  final String id;
  final String action;
  final String module;
  final String priority;
  final DateTime? dueDate;
  final String status;
  final String assignedTo;

  PendingAction({
    required this.id,
    required this.action,
    required this.module,
    required this.priority,
    this.dueDate,
    required this.status,
    required this.assignedTo,
  });

  factory PendingAction.fromJson(Map<String, dynamic> json) {
    return PendingAction(
      id: json['id']?.toString() ?? '',
      action: json['action'] as String,
      module: json['module'] as String,
      priority: json['priority']?.toString() ?? 'LOW',
      dueDate: json['dueDate'] != null ? DateTime.parse(json['dueDate'] as String) : null,
      status: json['status']?.toString() ?? 'PENDING',
      assignedTo: json['assignedTo'] as String,
    );
  }
}
